addons/purchase_stock/models/stock.py

```diff
@@ -212,6 +212,30 @@ def action_view_purchase(self):
 
         return result
 
+    def action_replenish(self):
+        if self.route_id == self.env.ref('purchase_stock.route_warehouse0_buy'):
+            supplier = self.product_id.with_company(self.company_id.id)._select_seller(
+                                        quantity=self.product_min_qty,
+                                        uom_id=self.product_uom)
+            supplier = supplier or self.product_id._prepare_sellers(False).filtered(
+                lambda s: not s.company_id or s.company_id == self.company_id
+            )[:1]
+
+            if not supplier and self.env['product.product'].check_access_rights('write'):
+                view = self.env.ref('purchase_stock.stock_warn_missing_vendor_form_view')
+                return {
```

Comment:
Seems to be overkill to create a wizard just to return an action. I would use a RedirectWarning instead 
https://github.com/odoo/odoo/blob/4de0ae434d1adc97fc1077b8d0f23c7e800ec161/odoo/exceptions.py#L42-L50